src/meta/src/hummock/manager/mod.rs

```diff
@@ -1149,10 +1167,72 @@ where
                 ..Default::default()
             },
         );
+
         let mut new_hummock_version = old_version;
-        new_hummock_version.id = new_version_id;
         new_version_delta.id = new_version_id;
+        new_hummock_version.id = new_version_id;
 
+        for (group_id, is_not_empty) in old_version_groups {
+            if !is_not_empty && !compaction_groups.contains_key(&group_id) {
+                let level_deltas = &mut new_version_delta
+                    .level_deltas
+                    .entry(group_id)
+                    .or_default()
+                    .level_deltas;
+                level_deltas.push(LevelDelta {
+                    delta_type: DeltaType::Groupdestroy as i32,
+                    level_idx: u32::MAX,
+                    l0_sub_level_id: u64::MAX,
+                    ..Default::default()
+                });
+                new_hummock_version.levels.remove(&group_id);
+            }
+        }
+        // these `group_id`s must be unique
+        for (
+            group_id,
+            CompactionGroup {
+                compaction_config, ..
+            },
+        ) in compaction_groups
+        {
+            if !new_hummock_version.levels.contains_key(&group_id) {
+                new_hummock_version
+                    .levels
+                    .try_insert(group_id, {
+                        let mut levels = vec![];
```

Comment:
Consider extract levels initialization, as we have some duplicate now.

Reply:
> Consider extract levels initialization, as we have some duplicate now.

fixed